### PR TITLE
handle high priority browsochrones requests via side channels.

### PR DESCRIPTION
**Pull request by mattwigway**
Previously the side channel would only open for old-style analyst requests, meaning that single point jobs would not complete when a regional job was running on the same graph. This fixes that and also cleans up the auto-shutdown code.

### CODE REVIEW

File: src/main/java/com/conveyal/r5/analyst/cluster/AnalystWorker.java [hunk 1]
```diff
@@ -825,6 +821,20 @@ private void logQueueStatus() {
         LOG.debug("Waiting tasks: high priority: {}, batch: {}", highPriorityExecutor.getQueue().size(), batchExecutor.getQueue().size());
     }
 
+    public static boolean isHighPriorityTask (GenericClusterRequest req) {
```
Comment (abyrd) [inline src/main/java/com/conveyal/r5/analyst/cluster/AnalystWorker.java]: Conditional-based polymorphism looks a bit suspicious to me. Couldn't this be handled with an isHighPriority method on GenericClusterRequest?

File: src/main/java/com/conveyal/r5/analyst/cluster/AnalystWorker.java [hunk 2]
```diff
@@ -300,15 +300,15 @@ public void run() {
...
                         this.handleOneRequest(t);
                     }));
 
             // Enqueue low-priority (batch) tasks; note that this may block anywhere in the process
             logQueueStatus();
-            tasks.stream().filter(t -> !(t instanceof AnalystClusterRequest) || ((AnalystClusterRequest) t).outputLocation != null)
+            tasks.stream().filter(t -> !isHighPriorityTask(t))
                 .forEach(t -> {
                     // attempt to enqueue, waiting if the queue is full
                     while (true) {
```
Comment (abyrd) [inline src/main/java/com/conveyal/r5/analyst/cluster/AnalystWorker.java]: This isn't part of the PR but looks pretty questionable to me. The thread sleeps when insertion is rejected? It seems like the executor should handle that kind of thing.

**Comment by mattwigway** [comment]
@abyrd should be good to go, please re-review.

**Comment by abyrd** [comment]
Great, I find this very readable after the changes! It looks like you added a log statement that was making the log too long, which caused the build to fail. I removed that and will merge.

**Pull request merged**